src/osemgrep/targeting/Glob_matcher.mli

```diff
@@ -0,0 +1,90 @@
+(*
+   AST and matching of a glob pattern against a path.
+   This is purely syntaxic: the file system is not accessed.
+*)
+
+(* The location of a pattern, for logging and troubleshooting. *)
+type loc = {
```

Comment:
I think this new type is justified because this `loc` contains the line of data it's referring to, which we quote in case of an error. Since it's often a pattern that comes from the command line (`--exclude foo`), we need to keep it around. It's acceptable because patterns are short but it would not be acceptable for locations in programs in general.